form.php

```diff
@@ -1,321 +1,336 @@
+    {
+        $middle_name_err = 'Length must be less than 200';
+    }
+
+    // To check error in employment.
+    if ( ! $valid->is_valid_name($_POST['employment']))
+    {
+        $employment_err = 'Only letters and white space allowed in Employment.';
+    }
+    else if ( ! $valid->valid_max_length($_POST['employment'], 30))
+    {
+        $employment_err = 'Length must be less than 30';
     }
 
     // To check error in last name.
-    if (empty($_POST['last_name'])) {
+    if ( ! $valid->is_empty($_POST['last_name']))
+    {
         $last_name_err = 'Last Name is required.';
-        $error++;
-     } else {
-
-        // Check if name only contains letters and whitespace
-        if ( ! preg_match('/^[a-zA-Z ]*$/', $last_name)) {
-          $last_name_err = 'Only letters and white space allowed.';
-          $error++;
-        }
+    }
+    else if ( ! $valid->is_valid_name($_POST['last_name']))
+    {
+        $last_name_err = 'Only letters and white space allowed.';
+    }
+    else if ( ! $valid->valid_max_length($_POST['last_name']))
+    {
+        $last_name_err = 'Length must be less than 200';
     }
 
     // To check error in date of birth.
-    if (empty($_POST['date_of_birth'])) {
+    if ( ! $valid->is_empty($_POST['date_of_birth']))
+    {
         $dob_err = 'Date of Birth is required.';
-        $error++;
-    } else {
-        $date = explode('-', $date_of_birth);
-
-        if (count($date) !== 3 || ! checkdate($date[1], $date[2], $date[0])){
-            $dob_err = 'Date of Birth is invalid.';
-        }
+    }
+    else if (! $valid->is_valid_date($_POST['date_of_birth']) )
+    {
+        $dob_err = 'Date of Birth is invalid.';
     }
 
     // To check error in pin.
-    if (empty($_POST['r_pin'])) {
-       $r_pin_err = 'This field is required.';
-       $error++;
-    } else if ( ! preg_match('/^[0-9]{6}$/', $r_pin)) {
+    if ( ! $valid->is_empty($_POST['r_pin']))
+    {
+        $r_pin_err = 'This field is required.';
+    }
+    else if ( ! $valid->is_valid_number($_POST['r_pin'],6))
+    {
         $r_pin_err = 'Invalid Pin Code.';
-        $error++;
     }
 
-    if ( ! preg_match('/^[0-9]{6}$/', $o_pin) && ! empty($o_pin)) {
+    if ( ! preg_match('/^[0-9]{6}$/', $_POST['o_pin']) && ! empty($_POST['o_pin']))
+    {
         $o_pin_err = 'Invalid Pin Code.';
         $error++;
     }
 
     // To check error in mobile no.
-    if (empty($_POST['r_phone'])){
+    if (! $valid->is_empty($_POST['r_phone']))
+    {
         $r_phone_err = 'This field is required.';
-        $error++;
-    } else if ( ! preg_match('/^[0-9]{10}$/', $r_phone)) {
+    }
+    else if ( ! $valid->is_valid_number($_POST['r_phone'], 10))
+    {
         $r_phone_err = 'Invalid Phone Number.';
-        $error++;
-   }
+    }
 
-    if ( ! preg_match('/^[0-9]{10}$/', $o_phone) && ! empty($o_phone)) {
+    if ( ! preg_match('/^[0-9]{10}$/', $_POST['o_phone']) && ! empty($_POST['o_phone']))
```

Comment:
empty should be the first check in the condition and can use !== '' instead of function call